pkg/tfbridge/provider_test.go

```diff
@@ -4535,3 +4535,1072 @@ func TestPlanResourceChangeStateUpgrade(t *testing.T) {
+			}
+		}
+	}`)
+	})
+
+	t.Run("unknown for list block prop", func(t *testing.T) {
+		testutils.Replay(t, provider, `
+	{
+		"method": "/pulumirpc.ResourceProvider/Create",
+		"request": {
+			"urn": "urn:pulumi:dev::teststack::ExampleResource::exres",
+			"properties":{
+				"__defaults":[],
+				"listBlockProps":["04da6b54-80e4-46f7-96ec-b56ff0331ba9"]
+			},
+			"preview":true
+		},
+		"response": {
+			"properties":{
+				"id":""
+			}
+		}
+	}`)
+	})
+
+	t.Run("unknown for list block prop collection", func(t *testing.T) {
+		testutils.Replay(t, provider, `
+	{
+		"method": "/pulumirpc.ResourceProvider/Create",
+		"request": {
+			"urn": "urn:pulumi:dev::teststack::ExampleResource::exres",
+			"properties":{
+				"__defaults":[],
+				"listBlockProps":"04da6b54-80e4-46f7-96ec-b56ff0331ba9"
+			},
+			"preview":true
+		},
+		"response": {
+			"properties":{
+				"id":"",
+				"listBlockProps":[null]
+			}
+		}
+	}`)
+	})
+
+	t.Run("unknown for nested list prop", func(t *testing.T) {
+		testutils.Replay(t, provider, `
+	{
+		"method": "/pulumirpc.ResourceProvider/Create",
+		"request": {
+			"urn": "urn:pulumi:dev::teststack::ExampleResource::exres",
+			"properties":{
+				"__defaults":[],
+				"nestedListProps":[["04da6b54-80e4-46f7-96ec-b56ff0331ba9"]]
+			},
+			"preview":true
+		},
+		"response": {
+			"properties":{
+				"id":"",
+				"nestedListProps":["04da6b54-80e4-46f7-96ec-b56ff0331ba9"]
+			}
+		}
+	}`)
+	})
+
+	t.Run("unknown for nested list block prop nested subprop", func(t *testing.T) {
+		testutils.Replay(t, provider, `
+	{
+		"method": "/pulumirpc.ResourceProvider/Create",
+		"request": {
+			"urn": "urn:pulumi:dev::teststack::ExampleResource::exres",
+			"properties":{
+				"__defaults":[],
+				"nestedListBlockProps":[{"nestedProps":[{"prop":"04da6b54-80e4-46f7-96ec-b56ff0331ba9"}]}]
+			},
+			"preview":true
+		},
+		"response": {
+			"properties":{
+				"id":"",
+				"nestedListBlockProps":[{"nestedProps":[{"prop":"04da6b54-80e4-46f7-96ec-b56ff0331ba9"}]}]
+			}
+		}
+	}`)
+	})
+
+	t.Run("unknown for nested list block nested prop", func(t *testing.T) {
+		testutils.Replay(t, provider, `
+	{
+		"method": "/pulumirpc.ResourceProvider/Create",
+		"request": {
+			"urn": "urn:pulumi:dev::teststack::ExampleResource::exres",
+			"properties":{
+				"__defaults":[],
+				"nestedListBlockProps":[{"nestedProps":["04da6b54-80e4-46f7-96ec-b56ff0331ba9"]}]
+			},
+			"preview":true
+		},
+		"response": {
+			"properties":{
+				"id":"",
+				"nestedListBlockProps":[{"nestedProps":"04da6b54-80e4-46f7-96ec-b56ff0331ba9"}]
+			}
+		}
+	}`)
+	})
+
+	t.Run("unknown for nested list block prop nested collection", func(t *testing.T) {
+		testutils.Replay(t, provider, `
+	{
+		"method": "/pulumirpc.ResourceProvider/Create",
+		"request": {
+			"urn": "urn:pulumi:dev::teststack::ExampleResource::exres",
+			"properties":{
+				"__defaults":[],
+				"nestedListBlockProps":[{"nestedProps":"04da6b54-80e4-46f7-96ec-b56ff0331ba9"}]
```

Comment:
This special case deserves a comment that an unknown is being transformed to a null in an attempt to avoid sending unknowns where TF does not support them. Still this probably is a bug as far as the provider's concerned. 

Without a comment it looks like it's asserting behavior we want.